lib/matplotlib/axes/_base.py

```diff
@@ -3603,15 +3604,30 @@ def invert_xaxis(self):
         """
         Invert the x-axis.
 
+        .. admonition:: Discouraged
+
+            The use of this method is discouraged.
+            Use `.Axes.set_xinverted` instead.
+
         See Also
         --------
-        xaxis_inverted
+        get_xinverted
         get_xlim, set_xlim
         get_xbound, set_xbound
         """
         self.xaxis.set_inverted(not self.xaxis.get_inverted())
 
+    set_xinverted = _axis_method_wrapper("xaxis", "set_inverted")
+    get_xinverted = _axis_method_wrapper("xaxis", "get_inverted")
     xaxis_inverted = _axis_method_wrapper("xaxis", "get_inverted")
+    if xaxis_inverted.__doc__:
```

Comment:
Why is this necessary?

Reply:
`__doc__` can be unset in "optimized" mode (https://docs.python.org/3/using/cmdline.html#cmdoption-OO).  There's already a few other places in the codebase with the same check.

Reply:
Sorry, still not following what this does or why...  Is this because we are building `__doc__` dynamically?

Reply:
Yes, in optimized mode the following line `xaxis_inverted.__doc__ += …” would raise and therefore must be guarded.